GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -9,17 +9,194 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+    
+    var seasons: [[GOTEpisode]] = GOTEpisode.allSeasons
+    
+    var filteredSeasons: [[GOTEpisode]] {
+        get {
+            guard let searchString = searchString else { return seasons }
+            
+            guard searchString != "" else { return seasons}
+            
+            if let scopeArrTitles = mySearchBar.scopeButtonTitles {
+                let currentIndex = mySearchBar.selectedScopeButtonIndex
+                let selectedStr = scopeArrTitles[currentIndex]
+                
+                return GOTEpisode.getFilteredResults(selectedStr: selectedStr, searchText: searchString)
+            }
+            
+            return seasons
+        }
+    }
+    
+    var searchString: String? = nil {
+        didSet {
+            self.myTableView.reloadData()
+        }
+    }
+    
+    @IBOutlet weak var myTableView: UITableView!
+    
+    @IBOutlet weak var mySearchBar: UISearchBar!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
+        configureTableView()
+        configureSearchBar()
     }
 
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
+    
+    func configureTableView() {
+        myTableView.dataSource = self
+        myTableView.delegate = self
+        myTableView.rowHeight = 80
+        myTableView.sectionHeaderHeight = 40
+        myTableView.tableFooterView = UIView()
+    }
+    
+    func configureSearchBar() {
+        mySearchBar.delegate = self
+    }
+    
+    
+
+
+}
+
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
+   
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return filteredSeasons[section].count
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 7
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let episode = filteredSeasons[indexPath.section][indexPath.row]
+       
+        let image: UIImage = (UIImage(named: episode.mediumImageID ))!
+        
+        if indexPath.section % 2 == 0 {
```

Comment:
great job making an if else statement to cover all of the cases instead of having a switch case with a meaningless default value